### ISSUE TITLE
e2e flake: SchedulerPredicates [Serial] validates MaxPods limit number of pods that are allowed to run [Slow]

### ISSUE THREAD

**Issue by mikedanese** [kubernetes/kubernetes#19681]
SchedulerPredicates [Serial] validates MaxPods limit number of pods that are allowed to run [Slow]

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:264 Expected : 325 to equal : 330

Flaked twice today.

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-reboot/10473/
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-reboot/10474/

**Comment by gmarek** [comment]
It flakes only in reboot suite, which means that disruptive tests somehow impact scheduler tests. My guess is that we don't wait for cluster is stable after some test.

I won't have time to fix this - I'll remove serial tests from the reboot suite until it's fixed.

cc @kubernetes/goog-control-plane @hongchaodeng @xiang90

**Comment by gmarek** [comment]
Actually a lot of tests are broken if run together with reboot. Reverting #19381 to unblock the build.

**Comment by gmarek** [comment]
It turned out not to be a flake, but real issue. Only failure of this test I was able to find is in http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/1486/, which reads:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:264 Expected error: <*errors.StatusError | 0xc208fd5080>: { ErrStatus: { TypeMeta: {Kind: "", APIVersion: ""}, ListMeta: {SelfLink: "", ResourceVersion: ""}, Status: "Failure", Message: "pods \"additional-pod\" not found", Reason: "NotFound", Details: { Name: "additional-pod", Group: "", Kind: "pods", Causes: nil, RetryAfterSeconds: 0, }, Code: 404, }, } pods "additional-pod" not found not to have occurred
```

I guess 404 means a backend problem.

**Comment by quinton-hoole** [comment]
Re-opening.  @gmarek is the test still failing occasionally?  If so, we need to fix something, either the system or the test.  It sounds like the system needs fixing.

**Comment by gmarek** [comment]
I think that this 404 was a GCE issue. I didn't investigate it thoroughly .

**Comment by quinton-hoole** [comment]
This test failed again yesterday:

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-slow/1534/

SchedulerPredicates [Serial] validates MaxPods limit number of pods that are allowed to run [Slow]

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:264 Expected : 330 to equal : 440

Looking through the code, I noticed this comment, which says that this test should not be running with other tests.  

```
	// This test verifies that max-pods flag works as advertised. It assumes that cluster add-on pods stay stable
	// and cannot be run in parallel with any other test that touches Nodes or Pods. It is so because to check
	// if max-pods is working we need to fully saturate the cluster and keep it in this state for few seconds.
	//
	// Slow PR #13315 (8 min)
```

I believe that this was the case before @ihmccreery re-organized test groupings recently.  Assigning to him to move this back to the right place.  Until then it will probably keep failing, and causing other tests to flake too (is this test saturates the cluster, so other tests running at the same time won't be able to schedule any pods for some period).

**Comment by quinton-hoole** [comment]
Oh wait, it is labelled as serial, so this should be fine - I'll dig further.

**Comment by quinton-hoole** [comment]
In the above test run, 9 other tests failed too.  One of the 4 nodes in the test cluster became unavailable in the middle of the test.  That explains why only 330 and not 440 pods were launched.  There is a separate issue tracking nodes becoming unavailable.  So closing this issue.

**Comment by davidopp** [comment]
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gke-serial/690/

**Comment by hongchaodeng** [comment]
Can you share the log? I can help with debugging.

**Comment by mikedanese** [comment]
@hongchaodeng can you see this: http://ci-test.k8s.io/kubernetes-e2e-gke-serial/690/

**Comment by hongchaodeng** [comment]
The scheduler log was empty when I downloaded (multiple times).

**Comment by hongchaodeng** [comment]
I just looked at the client log. It already looked like a hardware crash.

There were only 3 nodes while I thought there should be 4nodes, right?
```
FailedScheduling: pod (maxp-320) failed to fit in any node
fit failure on node (gke-jenkins-e2e-68d4e327-node-nhx4): Node didn't have enough resource: PodCount, requested: 1, used: 110, capacity: 110
fit failure on node (gke-jenkins-e2e-68d4e327-node-va8s): Node didn't have enough resource: PodCount, requested: 1, used: 110, capacity: 110
fit failure on node (gke-jenkins-e2e-68d4e327-node-x1rt): Node didn't have enough resource: PodCount, requested: 1, used: 110, capacity: 110
```

**Comment by wojtek-t** [comment]
@mikedanese - I'm afraid we don't have logs from master components from GKE runs...

**Comment by pwittrock** [comment]
This has been failing pretty consistently in the gce 1.2 soak cluster

kubernetes-soak-continuous-e2e-gce-1.2

**Comment by gmarek** [comment]
Failures of MaxPods is caused by #23670

**Comment by gmarek** [comment]
As it's caused by the Docker problem and it should be fixed when we start using Docker 1.10 I'm removing flake label. @dchen1107

**Comment by gmarek** [comment]
This test is stable now.

**Comment by davidopp** [comment]
This is failing deterministically in kubernetes-e2e-gce-serial.

http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-serial/1381/
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-serial/1380/
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-serial/1379/
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-serial/1378/
and so on.

**Comment by hongchaodeng** [comment]
I'm concerned that #26043 is highly relevant. Not sure why there is no response on that issue.
I will submit a patch to print error log to verify the case.

**Comment by hongchaodeng** [comment]
Submit a patch at #26496

**Comment by gmarek** [comment]
It was broken by image puller. Fixed by #26695
